### PR TITLE
State Update Delay configuration

### PR DESCRIPTION
**Pull request by sdlyy**
Resolves L2B-4184
Resolves L2B-4247

### CODE REVIEW

**Comment by sdlyy** [comment]
Prod and staging vercel builds fail since we have old API there ;p

File: packages/backend/src/peripherals/database/migrations/107_extend_finality.ts [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+If you find that something was incorrectly set up in the `up` function you
+should create a new migration file that fixes the issue.
+
+*/
+
+import { Knex } from 'knex'
+
+export async function up(knex: Knex) {
+  await knex.schema.alterTable('finality', function (table) {
+    table.integer('average_state_update')
```
Comment (maciekzygmunt) [inline packages/backend/src/peripherals/database/migrations/107_extend_finality.ts:18]: It should be nullable

**Review by maciekzygmunt — CHANGES_REQUESTED** [review]

File: packages/backend/src/modules/finality/FinalityIndexer.ts [hunk 2]
```diff
@@ -96,19 +96,47 @@ export class FinalityIndexer extends ChildIndexer {
...
     const from = to.add(-1, 'days')
 
-    const projectFinalityTimestamps =
-      await configuration.analyzer.getFinalityForInterval(from, to)
+    const inclusionDelays = await timeToInclusion.analyzeInterval(from, to)
 
-    if (!projectFinalityTimestamps) return
+    if (!inclusionDelays) {
+      return
+    }
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/backend/src/modules/finality/FinalityIndexer.ts:108]: Valid case but to calculate SUD for the given time range we need TTI for the same time range. The second analyzer (state update) only calculates the time to state update, and not state update delay

[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/backend/src/modules/finality/FinalityIndexer.ts:108]: This case will be handled as a part of extending this logic later on so each analyzer will return `Map<BlockNumber, TTI>` we can compare later on

File: packages/frontend/src/pages/scaling/finality/types.ts [hunk 3]
```diff
@@ -37,10 +37,17 @@ export interface ScalingFinalityViewEntry {
...
+    averageInSeconds: number
   }
   syncStatus: SyncStatus
 }
+
+export type FinalityDataTimings = {
+  minimumInSeconds: number | undefined
+  averageInSeconds: number
+  maximumInSeconds: number
+}
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/frontend/src/pages/scaling/finality/types.ts]: [`c10e7c5`](https://github.com/l2beat/l2beat/pull/3502/commits/c10e7c52f05ada6d4feb7802d15a95f6ae42e45d)

File: packages/backend/src/peripherals/database/migrations/107_extend_finality.ts (hunk 1 above)
Comment (sdlyy) [reply packages/backend/src/peripherals/database/migrations/107_extend_finality.ts]: https://github.com/l2beat/l2beat/pull/3502/commits/7967cf4c9752fbb77d209a183af1dde7a72ce05d

File: packages/frontend/src/components/table/FinalityDurationCell.tsx [hunk 4]
```diff
@@ -9,42 +10,47 @@ import { DurationCell } from './DurationCell'
 import { GrayedOut } from './GrayedOut'
 
 interface Props {
-  data: ScalingFinalityViewEntryData
+  scope: 'timeToInclusion' | 'stateUpdateDelay'
+  timings: Partial<FinalityDataTimings> &
+    Required<Pick<FinalityDataTimings, 'averageInSeconds'>>
+  syncStatus: SyncStatus
+  warning?: string
```
[the remark this answers is not in the document]
Comment (sdlyy) [reply packages/frontend/src/components/table/FinalityDurationCell.tsx]: https://github.com/l2beat/l2beat/pull/3502/commits/1c7bd078342f51281f14354a1820fff5d06e0f6a

**Pull request merged**